### DISCUSSION TITLE
How to make items persist in an item

### DISCUSSION THREAD

**Discussion by superhighlevel** [peter-kish/gloot#264] [Q&A]
First, thank you for making a great Inventory system.
I'm trying to make an indie game that has an inventory system like Escape From Tarkov(EFT). In this game, a backpack item can hold other items. I try to save ItemRef, if save the item here but the other function does not work as intended. I tried to work around it by making an Inventory Node and linking to that Item but this will makes save the file so much harder

**Comment by peter-kish** [comment]
There were some discussions about nested inventories in the past and there's an [open issue](https://github.com/peter-kish/gloot/issues/208) for that, but unfortunately, there's still no "official" support for that. Hopefully it will be possible to do once version 3.0 is out (still [work-in-progress](https://github.com/peter-kish/gloot/tree/dev_v3.0.0)). Until then, you can play around with extending `InventoryItem`, adding an `Inventory` member and making custom UI controls, but I haven't tried anything similar and I can't tell you how to implement it exactly nor if it will work ☹️

**Comment by superhighlevel** [comment]
Thank you for your response, I find a workaround for this case. It uses version 3.0, and makes a lot of modifications to make it work
Basically, I saved every item in Inventory, index, position, and item prototype,... and loaded/Added Items with that info when needed.
